### DISCUSSION TITLE
Protect routes without a brief flash of content

### DISCUSSION THREAD

**Discussion by callumok2004** [nextauthjs/next-auth#2667] [Help] [answered]
So I have a simple auth manager to protect routes but the one issue is this. When a user that is not logged in goes to route `/test`, they can see the content of that route for a brief second before being redirected. How can I overcome this?

```tsx
import { getSession } from 'next-auth/client';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

export default function AuthManager({ children }) {
    const [user, setUser] = useState(false as any),
        router = useRouter();

    useEffect(() => {
        async function session() {
            await setUser(await getSession());

            if (user) {
                if (router.asPath.startsWith("/auth")) router.push("/home");
            } else if (!router.asPath.startsWith("/auth")) {
                console.log(router)
                router.push("/auth");
            }
        }
        session();
    }, []);

    return !user && router.asPath.startsWith("/auth") ? children : <>
        <div style={{ width: "100vw", height: "100vh", background: "#0d0e11" }}></div>
        <script>window.location.href = "/"</script>
    </>;
}
```

**Comment by balazsorban44** [comment, marked as answer]
You can use getServerSideProps to avoid the flash by getting the session there. It might not be desired depending on your usecase, since it is blocking and will result in a longer TTFB.

Instead of creating your AuthManager, you could use the built-in Provider from `next-auth/client`

https://next-auth.js.org/getting-started/client#provider

**Comment by NyakoFox** [comment]
Hello, is there a way to do this with Next 13's app router, since it doesn't have `getServerSideProps`?

**Reply by GermanJablo** [reply]
Could you solve it?
